### ISSUE TITLE
OOM error of training

### ISSUE THREAD

**Issue by DrChiZhang** [CIntellifusion/GeometryForcing#3]
Hi, Thank you for the amazing work! 
I have tried to train this model with 8 * A100 with 80GB VRAM and it encounters OOM error even with very small batch size 4.  
While, in you paper, seems batch size 8 is used. any clue for this issue? 

sweep_config: {}
config: {'experiment': {'debug': '${debug}', 'tasks': ['training'], 'num_nodes': 1, 'training': {'precision': '16-mixed', 'compile': False, 'lr': 8e-06, 'batch_size': 2, 'max_epochs': 24, 'max_steps': -1, 'max_time': None, 'data': {'num_workers': 11, 'shuffle': False}, 'optim': {'accumulate_grad_batches': 5, 'gradient_clip_val': 1.0}, 'checkpointing': {'every_n_train_steps': None, 'every_n_epochs': 1, 'train_time_interval': None, 'enable_version_counter': False, 'save_top_k': -1}}, 'validation': {'precision': '16-mixed', 'compile': False, 'batch_size': 1, 'val_every_n_step': 1.0, 'val_every_n_epoch': 1, 'num_sanity_val_steps': 1, 'limit_batch': 1.0, 'data': {'num_workers': 0, 'shuffle': False}, 'inference_mode': True}, 'test': {'precision': '16-mixed', 'compile': False, 'batch_size': 4, 'limit_batch': 1.0, 'data': {'num_workers': 0, 'shuffle': False}, 'inference_mode': True}, 'find_unused_parameters': False, 'reload_dataloaders_every_n_epochs': 1, 'ema': {'enable': True, 'decay': 0.9999, 'validate_original_weights': False}, '_name': 'video_generation'}, 'dataset': {'debug': '${debug}', 'save_dir': 'data/real-estate-10k', 'data_mean': [[[0.577]], [[0.517]], [[0.461]]], 'data_std': [[[0.249]], [[0.249]], [[0.268]]], 'latent': {'enable': False, 'type': 'pre_sample', 'suffix': None, 'downsampling_factor': [1, 8], 'num_channels': 4}, 'resolution': 256, 'observation_shape': [3, '${dataset.resolution}', '${dataset.resolution}'], 'max_frames': 16, 'n_frames': '${dataset.max_frames}', 'context_length': 4, 'frame_skip': 10, 'filter_min_len': None, 'external_cond_dim': 16, 'external_cond_stack': False, 'external_cond_processing': None, 'preload': False, 'subdataset_size': 10000, 'num_eval_videos': 100, 'maximize_training_data': True, 'augmentation': {'frame_skip_increase': 1, 'horizontal_flip_prob': 0.5, 'reverse_prob': 0.5, 'back_and_forth_prob': 0.1}, '_name': 'realestate10k'}, 'algorithm': {'debug': '${debug}', 'lr': '${experiment.training.lr}', 'backbone': {'name': 'u_vit3d_pose', 'channels': [128, 256, 576, 1152], 'emb_channels': 1024, 'patch_size': 2, 'block_types': ['ResBlock', 'ResBlock', 'TransformerBlock', 'TransformerBlock'], 'block_dropouts': [0.0, 0.0, 0.1, 0.1], 'num_updown_blocks': [3, 3, 6], 'num_mid_blocks': 20, 'num_heads': 9, 'pos_emb_type': 'rope', 'use_checkpointing': [False, False, False, True], 'conditioning': {'dim': None}, 'external_cond_dropout': 0.1, 'use_fourier_noise_embedding': True}, 'x_shape': '${dataset.observation_shape}', 'max_frames': '${dataset.max_frames}', 'n_frames': '${dataset.n_frames}', 'frame_skip': '${dataset.frame_skip}', 'context_frames': '${dataset.context_length}', 'latent': '${dataset.latent}', 'data_mean': '${dataset.data_mean}', 'data_std': '${dataset.data_std}', 'external_cond_dim': '${dataset.external_cond_dim}', 'external_cond_stack': '${dataset.external_cond_stack}', 'external_cond_processing': '${dataset.external_cond_processing}', 'compile': 'true_without_ddp_optimizer', 'weight_decay': 0.01, 'optimizer_beta': [0.9, 0.99], 'lr_scheduler': {'name': 'constant_with_warmup', 'num_warmup_steps': 10000, 'num_training_steps': 550000}, 'noise_level': 'random_independent', 'uniform_future': {'enabled': False}, 'fixed_context': {'enabled': False, 'indices': None, 'dropout': 0}, 'variable_context': {'enabled': False, 'prob': 0, 'dropout': 0}, 'chunk_size': -1, 'scheduling_matrix': 'full_sequence', 'replacement': 'noisy_scale', 'diffusion': {'is_continuous': True, 'timesteps': 1000, 'beta_schedule': 'cosine_simple_diffusion', 'schedule_fn_kwargs': {'shift': 1.0, 'shifted': 0.125, 'interpolated': False}, 'use_causal_mask': False, 'clip_noise': 20.0, 'objective': 'pred_v', 'loss_weighting': {'strategy': 'sigmoid', 'snr_clip': 5.0, 'cum_snr_decay': 0.9, 'sigmoid_bias': -1.0}, 'sampling_timesteps': 50, 'ddim_sampling_eta': 0.0, 'reconstruction_guidance': 0.0, 'training_schedule': {'name': 'cosine', 'shift': 0.125}, 'precond_scale': 0.125}, 'vae': {'pretrained_path': None, 'pretrained_kwargs': {}, 'use_fp16': True, 'batch_size': 2}, 'checkpoint': {'reset_optimizer': False, 'strict': True}, 'tasks': {'prediction': {'enabled': True, 'history_guidance': {'name': 'conditional'}, 'keyframe_density': None, 'sliding_context_len': None}, 'interpolation': {'enabled': False, 'history_guidance': {'name': 'conditional'}, 'max_batch_size': None}}, 'logging': {'deterministic': 0, 'loss_freq': 100, 'grad_norm_freq': 100, 'max_num_videos': 256, 'n_metrics_frames': None, 'metrics': ['fvd', 'is', 'fid', 'lpips', 'mse', 'ssim', 'psnr'], 'metrics_batch_size': 16, 'sanity_generation': False, 'raw_dir': None}, 'camera_pose_conditioning': {'normalize_by': 'first', 'bound': None, 'type': 'ray_encoding'}, 'alignment': {'alignment_coeff': 0.1, 'encoder_type': 'vggt', 'apply_unnormalize_recon': True, 'alignment_context_length': 16, 'encoder_info': [24, 512, 512], 'mid_channels': 128, 'latents_info': 2}, '_name': 'dfot_geometry_forcing'}, 'debug': False, 'wandb': {'entity': 'zhangchi0118-huawei', 'project': 'dfot', 'mode': 'offline'}, 'resume': None, 'load': '/data/bucket-embodied-data-wulan1/MunichResearchCenter/pretraineded_model/GeoForcing/DFoT_16f_state_dict.ckpt', 'name': 'RE10k', '_wandb': {}}

**Comment by CIntellifusion** [comment]
global batch_size = batch_size x accumulation_steps , you can try reduce the batch_size.

**Comment by CIntellifusion** [comment]
As mentioned in paper, there wouldn't be OOM issue with GPU that has 80GB or larger memory
